compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -856,6 +856,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             format!("impl {}", all_bounds_str),
             Applicability::MaybeIncorrect,
         );
+        err.note("the type of T is chosen by the caller");
```

Comment:
This is going to be wrong if the parameter isn't named `T`.

```suggestion
        err.note(format!("the type of `{}` is chosen by the caller", expected_ty_as_param.name));
```